src/Cmfd.cpp

```diff
@@ -1788,3 +1822,58 @@ void Cmfd::updateBoundaryFlux(Track** tracks, FP_PRECISION* boundary_flux,
     }
   }
 }
+
+
+/**
+ * @brief Zero the surface currents for each mesh cell and energy group.
+ */
+void Cmfd::zeroSurfaceCurrents() {
+
+  #pragma omp parallel for schedule(guided)
+  for (int r=0; r < _num_x*_num_y; r++) {
+    for (int s=0; s < 8; s++) {
+      for (int e=0; e < _num_moc_groups; e++)
+        _surface_currents(r*8+s,e) = 0.0;
+    }
+  }
+
+  return;
+}
+
+
+/**
+ * @brief 
+ * @param curr_segment the current Track segment
+ * @param current the current to tally
+ * @param fwd boolean indicating direction of integration along segment
+ * @param e the MOC energy group index
+ */
+void Cmfd::tallySurfaceCurrent(segment* curr_segment, 
+                               FP_PRECISION current, bool fwd, int e) {
+
+  if (curr_segment->_cmfd_surface_fwd != -1 && fwd){
```

Comment:
The segments still have a ``_cmfd_surface_fwd`` and ``_cmfd_surface_bwd`` attribute. I think to compromise on computational efficiency and code structure, I should move the loop over energy groups and polar angles from the ``Solver::tallySurfaceCurrent(...)`` routine to here. That way the MOC ``Solver`` class is agnostic to CMFD-specific computations, like which segments intersect mesh surfaces and which do not. What do you think?